src/app/services/confirmServiceConfig.js

```diff
@@ -0,0 +1,295 @@
+    title: 'Post password-reset URL',
+    description: 'Where you want to redirect users after they have reset their password. It is usually the DfE Sign-in home page.',
+    changeLink: 'service-configuration?action=amendChanges#postResetUrl-form-group',
+    displayOrder: 2,
+  },
+  redirectUris: {
+    title: 'Redirect URL',
+    description: 'Where you want to redirect users after they have authenticated.',
+    changeLink: 'service-configuration?action=amendChanges#redirect_uris-form-group',
+    displayOrder: 3,
+  },
+  postLogoutRedirectUris: {
+    title: 'Logout redirect URL',
+    description: 'Where you want to redirect users after they log out of a service.',
+    changeLink: 'service-configuration?action=amendChanges#post_logout_redirect_uris-form-group',
+    displayOrder: 4,
+  },
+  responseTypes: {
+    title: 'Response types',
+    description: 'A value that determines the authentication flow.',
+    changeLink: 'service-configuration?action=amendChanges#response_types-form-group',
+    displayOrder: 5,
+  },
+  grantTypes: {
+    title: 'Grant types',
+    description: 'Grant types placeholder description.',
+    changeLink: 'service-configuration?action=amendChanges#clientSecret-form-group',
+    displayOrder: 6,
+  },
+  apiSecret: {
+    title: 'API Secret',
+    description: 'A value that is created automatically by the system and acts as a password for the DfE Sign-in public API.',
+    changeLink: 'service-configuration?action=amendChanges#apiSecret-form-group',
+    displayOrder: 7,
+  },
+};
+
+const getServiceConfigMapping = (key, sid) => {
+  const mapping = { ...serviceConfigChangesSummaryDetails[key] };
+  if (mapping) {
+    mapping.changeLink = `/services/${sid}/${mapping.changeLink}`;
+  }
+  return mapping;
+};
+
+const getAddedAndRemovedValues = (oldValue, newValue) => {
+  let addedValues = [];
+  let removedValues = [];
+
+  if (Array.isArray(oldValue) && Array.isArray(newValue)) {
+    addedValues = newValue.filter((v) => !oldValue.includes(v));
+    removedValues = oldValue.filter((v) => !newValue.includes(v));
+  } else if (typeof oldValue === 'string' && typeof newValue === 'string') {
+    if (oldValue !== newValue) {
+      addedValues = newValue === '' ? [] : [newValue];
+      removedValues = oldValue === '' ? [] : [oldValue];
+    }
+  }
+
+  return { addedValues, removedValues };
+};
+
+const createFlattenedMappedServiceConfigChanges = (serviceConfigurationChanges, sid) => Object.entries(serviceConfigurationChanges).map(([key, value]) => {
+  const mapping = getServiceConfigMapping(key, sid);
+  if (mapping) {
+    const { oldValue, newValue } = value;
+    const { addedValues, removedValues } = getAddedAndRemovedValues(oldValue, newValue);
+
+    return {
+      ...value,
+      ...mapping,
+      addedValues,
+      removedValues,
+    };
+  }
+  return value;
+});
+
+const buildCurrentServiceModel = async (req) => {
+  const service = await getServiceById(req.params.sid, req.id);
```

Comment:
can you please add some logging logger.info or logger.debug to record what user has selected on submit